packages/ui/src/components/EasySetup/IdentitySetIdentity.tsx

```diff
@@ -0,0 +1,127 @@
+import { Grid } from '@mui/material'
+import { styled } from '@mui/material/styles'
+import { SubmittableExtrinsic } from '@polkadot/api/types'
+import { ISubmittableResult } from '@polkadot/types/types'
+import { useCallback, useEffect, useState } from 'react'
+import { useApi } from '../../contexts/ApiContext'
+import { TextFieldStyled } from '../library'
+import { useIdentity } from '../../hooks/useIdentity'
+
+interface Props {
+  className?: string
+  from: string
+  onSetExtrinsic: (ext?: SubmittableExtrinsic<'promise', ISubmittableResult>) => void
+}
+
+interface IdentityFields {
+  display: string | undefined
+  legal: string | undefined
+  web: string | undefined
+  riot: string | undefined
+  email: string | undefined
+  image: string | undefined
+  twitter: string | undefined
+}
+
+const getRawOrNone = (val: string | undefined) => {
+  return val
+    ? {
+        Raw: val
+      }
+    : { none: null }
+}
+const getExtrinsicsArgs = (identity: IdentityFields) => {
+  const { legal, display, email, image, riot, twitter, web } = identity
+
+  return {
+    additional: [],
+    display: getRawOrNone(display),
+    legal: getRawOrNone(legal),
+    web: getRawOrNone(web),
+    riot: getRawOrNone(riot),
+    email: getRawOrNone(email),
+    pgpFingerprint: null,
+    image: getRawOrNone(image),
+    twitter: getRawOrNone(twitter)
+  }
+}
+
+const IdentitySetIdentity = ({ className, onSetExtrinsic, from }: Props) => {
+  const { api, chainInfo } = useApi()
+  const [identity, setIdentity] = useState<IdentityFields | undefined>()
+  const chainIdentity = useIdentity(from)
+
+  useEffect(() => {
+    if (chainIdentity) {
+      const { display, email, legal, web, riot, twitter, image } = chainIdentity
+      setIdentity({
+        display,
+        legal,
+        web,
+        riot,
+        email,
+        image,
+        twitter
+      })
+    } else {
+      setIdentity({
+        display: undefined,
+        legal: undefined,
+        web: undefined,
+        riot: undefined,
+        email: undefined,
+        image: undefined,
+        twitter: undefined
+      })
+    }
```

Comment:
It could be also moved to useIdentity hook

Reply:
Why? `useIdentity` is only there to fetch the identity from the chain. This is a hook that is re-used for other needs (the `AccountDisplay`)